src/hermes_claude_agent_sdk/parity/canonical.py

```
@@ -0,0 +1,319 @@
+# Catalog references and packet metadata are intentionally bounded.  These
+# limits are not a serialization substitute; they stop accidental inclusion
+# of a transcript, exception, or other unbounded value before hashing.
+MAX_STRING_LENGTH = 4096
+MAX_COLLECTION_LENGTH = 4096
+
+
+def _is_control_bearing(value: str) -> bool:
+    return _CONTROL_RE.search(value) is not None
+
+
+def validate_identifier(value: Any, *, field: str = "identifier", max_length: int = 256) -> str:
```

Ruff (0.16.3)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)